### PR TITLE
Various changes from reviews

### PR DESCRIPTION
**Pull request by StefanGreffenius**

### CODE REVIEW

File: lib/rocketfuel_api/service.rb [hunk 1]
```diff
@@ -19,12 +19,13 @@ def name
     end
   end
 
-  def plural_name
-    name + 's'
-  end
-
   def uri_suffix
-    plural_name
+    file_path = File.join(RocketfuelApi.root, 'lib/config/endpoints_for_services.yaml')
```
Comment (tsujigiri) [inline lib/rocketfuel_api/service.rb]: You can write this as `RocketfuelApi.root.join('lib', 'config', 'endpoints_for_services.yaml')`.

File: lib/rocketfuel_api/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module RocketfuelApi
-  VERSION = '0.1.1'
+  VERSION = '0.1.2'
```
Comment (tsujigiri) [inline lib/rocketfuel_api/version.rb]: Updating the version in the branch is usually not a good idea. Especially if more than one branch is open at the same time. I usually update them in the master branch after merging.

**Review by tsujigiri — APPROVED** [review]
:woman_astronaut:

File: lib/rocketfuel_api/version.rb (hunk 2 above)
Comment (StefanGreffenius) [reply lib/rocketfuel_api/version.rb]: All right, good to know.

File: lib/rocketfuel_api/service.rb [hunk 3]
```diff
@@ -42,8 +43,6 @@ def get_all(params = {})
   end
 
   def create(route_params = {}, body = {})
```
Comment (guigs) [inline lib/rocketfuel_api/service.rb:45]: Should't we remove the `create` and `update` methods?

File: lib/rocketfuel_api/service/advertisement_ad_assignment.rb [hunk 4]
```diff
@@ -2,12 +2,6 @@
 
 class RocketfuelApi::Service::AdvertisementAdAssignment < RocketfuelApi::Service
   def initialize(connection)
```
Comment (guigs) [inline lib/rocketfuel_api/service/advertisement_ad_assignment.rb:4]: now we can remove the initialize and keep all services classes empty

File: lib/rocketfuel_api/service.rb [hunk 5]
```diff
@@ -19,12 +19,13 @@ def name
...
   end
 
-  def plural_name
-    name + 's'
-  end
-
   def uri_suffix
-    plural_name
+    file_path = RocketfuelApi.root.join('lib', 'config', 'endpoints_for_services.yaml')
+    endpoints = YAML.load_file(file_path)
```
Comment (guigs) [inline lib/rocketfuel_api/service.rb:24]: Great! Would be good if we could somehow cache the endpoints hash so we don't need to load yml multiple times if we are using more than 1 service.

File: lib/rocketfuel_api/service.rb (hunk 3 above)
Comment (tsujigiri) [reply lib/rocketfuel_api/service.rb:45]: Why? Shouldn't an API client have them?

Comment (guigs) [reply lib/rocketfuel_api/service.rb:45]: Hum... before all services were setting read_only = true, so none could run `create` or `update`. So I thought we don't need them. But for some services create/update really doesn't make sense, for example for reporting. Should we add read_only logic back and set read_only = false for the ones that can create/updage?

Comment (tsujigiri) [reply lib/rocketfuel_api/service.rb:45]: The `read_only` part was something we copied over from the `appnexusapi` gem and deemed not necessary. [The API](https://api.rocketfuel.com/doc/2016/index.html) has endpoints to create and update things. It's not that well structured, but if something doesn't have a create or update endpoint, the API will tell you anyway. So, I'd just leave this in there.

**Pull request merged**